### PR TITLE
JAVA-1543: Update Contributing guide for DSE driver.

### PR DESCRIPTION
**Pull request by newkek**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -5,8 +5,7 @@
 Before starting to work on something, please comment in JIRA or ask on the mailing list
 to make sure nobody else is working on it.
 
-If your fix applies to multiple branches, base your work on the lowest active branch. The DSE driver uses
```
Comment (adutra) [inline CONTRIBUTING.md]: Why suppress this sentence?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -24,16 +23,12 @@ Legend:
 
 - new features are developed on "minor" branches such as `1.x`, where minor releases (ending in `.0`) happen.
 - bugfixes go to "patch" branches such as `1.1.x` and `1.2.x`, where patch releases (ending in `.1`, `.2`...) happen.
-- patch branches are regularly merged to the right (`1.1.x` to `1.2.x`) and to the bottom (`1.2.x` to `1.x`) so that
```
Comment (adutra) [inline CONTRIBUTING.md]: Same here, why suppress the fragment `right (`1.1.x` to `1.2.x`) and to the` ?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -24,16 +23,12 @@ Legend:
...
+  applied to newer versions too.
 
-The current active versions are 1.1 and 1.2. Therefore:
-
-- if you're fixing a bug on a feature that existed since 1.1, target `1.1.x`. Your changes will be available in future
-  1.1 and 1.2 patch versions.
-- if you're fixing a bug on a 1.2-only feature, target `1.2.x`. Your changes will be available in a future 1.2 patch
-  version.
-- if you're adding a new feature, target `1.x`. Your changes will be available in the upcoming 1.3.0.
+Consequently, the branch having the highest major + minor version (in the format `x.x.x`)
```
Comment (adutra) [inline CONTRIBUTING.md]: Why change this paragraph? I would rather update the version numbers and keep the global shape intact, to match what we have in the OSS driver.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -54,14 +49,14 @@ in-depth knowledge of the codebase.
 We use IntelliJ IDEA with the default formatting options, with one exception: check
 "Enable formatter markers in comments" in Preferences > Editor > Code Style.
 
-Please format your code and organize imports before submitting your changes.
+Please format your code and optimize imports before submitting your changes.
```
Comment (adutra) [inline CONTRIBUTING.md]: Again, we have "organize"in the OSS driver, why change it here to "optimize"?

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -71,16 +66,26 @@ mvn test -Pshort
...
+build.
+
 Two Maven properties control its execution:
 
-- `cassandra.version`: the Cassandra version. This has a default value in the root POM,
+- `cassandra.version`: the **DSE** version. This has a default value in the root POM,
   you can override it on the command line (`-Dcassandra.version=...`).
 - `ipprefix`: the prefix of the IP addresses that the Cassandra instances will bind to (see
   below). This defaults to `127.0.1.`.
+- `cassandra.directory`: The directory containing DSE's source project, in which CCM
```
Comment (adutra) [inline CONTRIBUTING.md]: This could be useful in the OSS driver file too.

File: CONTRIBUTING.md (hunk 1 above)
Comment (newkek) [inline CONTRIBUTING.md]: The previous contributing guide assumed we had multiple minor versions in active development, which is not correct anymore if I'm not mistaken. 

Here it explains that if you're working on a bug fix that can apply to multiple of the active minor, base it off of the lowest active minor. So I think it's not useful anymore

File: CONTRIBUTING.md (hunk 4 above)
Comment (newkek) [inline CONTRIBUTING.md]: Ah I thought I'd match with the IntelliJ option name to make it clearer what needs to be done

File: CONTRIBUTING.md (hunk 1 above)
Comment (tolbertam) [inline CONTRIBUTING.md]: I think we will still have multiple active branches, but it's less likely that we will be doing releases for those branches.  In past we were more apt to do patch releases even if there were changes that were more than bug fixes.  Now we are more likely to do minor releases, and only do patches for critical fixes.  However, it's not really clear to users what branches may be 'active'.  

I wonder if it is ok to remove this guidance and have users contribute pull requests on the primary branch.  We can then pull those changes back into older branches if they are applicable (although that is less likely with our change in strategy).  Either that or we should document what the actively maintained branches are, and keep the existing guidance here.

Comment (newkek) [inline CONTRIBUTING.md]: > I think we will still have multiple active branches, but it's less likely that we will be doing releases for those branches 

> I wonder if it is ok to remove this guidance and have users contribute pull requests on the primary branch. We can then pull those changes back into older branches

I don't understand, why would we pull changes back into older branches if we don't release them? (unless it's a critical bug that needs to be applied on older versions)
In the way I understand it currently, there's a branch for bug fixes (latest `x.x.x`), where users should point their PRs to, and a branch for new features (latest `x.x`) that users should target for new features, no?

Comment (tolbertam) [inline CONTRIBUTING.md]: > unless it's a critical bug that needs to be applied on older versions

Exactly, that should be the only reason we commit changes to the non-primary branch.

I think we either need to document what the 'active branches' are, or remove this guidance about doing work on the lowest active branch.  In the latter case, the user can just open up PRs on the primary branch and we can be responsible for pulling them back.

**Pull request merged**